spec/services/music_import_service_spec.rb

```diff
@@ -138,6 +141,163 @@
+            selection_id: 15_930,
+            file_path: "cd-1",
+            file_name: "cd-1_1.ra",
+            file_note: "A different file",
+            entry_id: "blabla",
+            selection_title: "My Selection",
+            selection_alt_title: nil,
+            selection_note: "Paul Jacobs, piano",
+            recording_id: 15
+          ),
+          MusicImportService::RecordingCollector::AudioFile.new(
+            id: 54_204,
+            selection_id: 15_929,
+            file_path: "cd-1",
+            file_name: "cd-1_1.ra",
+            file_note: "First File",
+            entry_id: "blabla",
+            selection_title: "Week 1",
+            selection_alt_title: nil,
+            selection_note: "Paul Jacobs, piano",
+            recording_id: 14
+          )
+        ]
+        selections = [
+          MusicImportService::RecordingCollector::Selection.new(
+            id: 15_929,
+            course_nums: ["mus204"]
+          )
+        ]
+        allow(importer.recording_collector).to receive(:audio_files).with(recording).and_return(audio_files)
+        allow(importer.recording_collector).to receive(:courses_for_selections).with([15_929]).and_return(selections)
+        allow(importer.recording_collector).to receive(:courses_for_selections).with([15_930]).and_return([])
+        prerequisite_collector = instance_double(MusicImportService::RecordingCollector)
+        allow(importer.recording_collector).to receive(:with_recordings_query).and_return(prerequisite_collector)
+        allow(importer.recording_collector).to receive(:audio_files).with(recording2).and_return(audio_files)
+        allow(importer.recording_collector).to receive(:prerequisite_recordings_query).with([15])
+        allow(prerequisite_collector).to receive(:recordings).and_return([recording2])
+
+        importer.ingest_recording(recording)
+        recording = Valkyrie::MetadataAdapter.find(:indexing_persister).query_service.find_all_of_model(model: ScannedResource)
+        expect(recording.length).to eq 1
+        expect(recording.first.member_ids.length).to eq 1
+        expect(recording.first.local_identifier).to eq ["15"]
+        members = Wayfinder.for(recording.first).members
+
+        expect(members.length).to eq 1
+        expect(members.first.title).to eq ["A different file"]
+        expect(members.first.original_file.original_filename).to eq ["cd-1_1.wav"]
+
+        playlists = Valkyrie::MetadataAdapter.find(:indexing_persister).query_service.find_all_of_model(model: Playlist)
+        expect(playlists.length).to eq 2
+        expect(playlists.flat_map(&:title)).to contain_exactly "Week 1", "My Selection"
+        playlist = playlists.find { |x| x.title.first == "Week 1" }
+        members = Wayfinder.for(playlist).members
+        expect(members.first.label).to eq ["First File"]
+        expect(members.first.local_identifier).to eq ["54204"]
+      end
+    end
+    context "when a dependent is a fake playlist" do
```

Comment:
I don't really understand what this means. what is a dependent exactly? and what is the relationship here?